### PR TITLE
deps: Bump `com_github_curl` -> 7.80.0

### LINKED ISSUE #18956: Newer release available `com_github_curl`: curl-7_80_0 (current: curl-7_79_1)
Package Name: com_github_curl
Current Version: curl-7_79_1@2021-09-22 06:19:51
Available Version: curl-7_80_0@2021-11-10 06:33:35
Upstream releases: https://github.com/curl/curl/releases

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue] Fix #18956
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -805,8 +805,8 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "curl",
         project_desc = "Library for transferring data with URLs",
         project_url = "https://curl.haxx.se",
-        version = "7.79.1",
-        sha256 = "370b11201349816287fb0ccc995e420277fbfcaf76206e309b3f60f0eda090c2",
+        version = "7.80.0",
+        sha256 = "ed936c0b02c06d42cf84b39dd12bb14b62d77c7c4e875ade022280df5dcc81d7",
```
Comment (moderation) [inline bazel/repository_locations.bzl:809]: `dab997c9b08cb4a636a03f2f7f985eaba33279c1c52692430018fae4a4878dc7`

**Review by moderation — CHANGES_REQUESTED** [review]

File: bazel/repository_locations.bzl (hunk 1 above)
Comment (phlax) [reply bazel/repository_locations.bzl:809]: wierd im still getting `ed936c0` with

```
curl -Ls https://github.com/curl/curl/releases/download/curl-7_78_0/curl-7.78.0.tar.gz | sha256sum
```

Comment (phlax) [reply bazel/repository_locations.bzl:809]: my bad - wrong version! ill update

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**